ktor-client/ktor-client-core/js/src/io/ktor/client/plugins/websocket/JsWebSocketSession.kt

```diff
@@ -42,21 +54,17 @@ internal class JsWebSocketSession(
             callback = {
                 val event = it.unsafeCast<MessageEvent>()
 
-                launch {
-                    val data = event.data
-
-                    val frame: Frame = when (data) {
-                        is ArrayBuffer -> Frame.Binary(false, Int8Array(data).unsafeCast<ByteArray>())
-                        is String -> Frame.Text(data)
-                        else -> {
-                            val error = IllegalStateException("Unknown frame type: ${event.type}")
-                            _closeReason.completeExceptionally(error)
-                            throw error
-                        }
+                val frame: Frame = when (val data = event.data) {
+                    is ArrayBuffer -> Frame.Binary(false, Int8Array(data).unsafeCast<ByteArray>())
+                    is String -> Frame.Text(data)
+                    else -> {
+                        val error = IllegalStateException("Unknown frame type: ${event.type}")
+                        _closeReason.completeExceptionally(error)
+                        throw error
                     }
-
-                    _incoming.trySend(frame).isSuccess
                 }
+
+                _incoming.trySend(frame)
```

Comment:
Should we check the operation result here?

Reply:
looks like no, as channel is unlimited and closed based on events from socket, so if channel is closed, we are already failing. Looks like it was before `offer` and just replaced with auto migration from coroutines.

But, if channel is closed externally (by user), we can't subscribe to it, but we can handle it here. 
Not sure, if such operations are supported in other WebSocketSession implementation. Will check it

Reply:
Looks like no other implementations except `test-host` implementation handles channels close events, so let's leave it like this for now in this PR